engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/ManagementServiceAsyncOperationsTest.java

```diff
@@ -384,6 +391,179 @@ public void testSetJobsRetryAsyncWithNegativeRetries() throws Exception {
+  }
+
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_ACTIVITY)
+  @Test
+  public void shouldSetJobDueDateOnJobRetryAsyncByProcessInstanceIds() {
+    //given
+    Date newDueDate = new Date(0);
+
+    //when
+    Batch batch = managementService.setJobRetriesByProcessAsync(RETRIES)
+        .processInstanceIds(processInstanceIds)
+        .dueDate(newDueDate).executeAsync();
+    completeSeedJobs(batch);
+    List<Exception> exceptions = executeBatchJobs(batch);
+
+    // then
+    assertThat(exceptions).hasSize(0);
+    for (String id : processInstanceIds) {
+      Job job = managementService.createJobQuery().processInstanceId(id).singleResult();
+      assertThat(job.getRetries()).isEqualTo(RETRIES);
+      assertThat(job.getDuedate()).isEqualTo(newDueDate);
+    }
+  }
+
+  @Test
+  public void shouldSetJobDueDateOnJobRetryAsyncByProcessInstanceQuery() {
+    //given
+    ProcessInstanceQuery query = runtimeService.createProcessInstanceQuery();
+    Date newDueDate = new Date(0);
+
+    //when
+    Batch batch = managementService.setJobRetriesByProcessAsync(RETRIES).processInstanceQuery(query).dueDate(newDueDate).executeAsync();
+    completeSeedJobs(batch);
+    List<Exception> exceptions = executeBatchJobs(batch);
+
+    // then
+    assertThat(exceptions).hasSize(0);
+    for (String id : ids) {
+      Job jobResult = managementService.createJobQuery().jobId(id).singleResult();
+      assertThat(jobResult.getRetries()).isEqualTo(RETRIES);
+      assertThat(jobResult.getDuedate()).isEqualTo(newDueDate);
+    }
+  }
+
+
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_ACTIVITY)
+  @Test
+  public void shouldSetJobDueDateOnJobRetryAsyncByHistoricProcessInstanceQuery() {
+    //given
+    HistoricProcessInstanceQuery historicProcessInstanceQuery =
+        historyService.createHistoricProcessInstanceQuery();
+
+    Date newDueDate = new Date(0);
+
+    //when
+    Batch batch = managementService.setJobRetriesByProcessAsync(RETRIES)
+        .historicProcessInstanceQuery(historicProcessInstanceQuery)
+        .dueDate(newDueDate).executeAsync();
+    completeSeedJobs(batch);
+    List<Exception> exceptions = executeBatchJobs(batch);
+
+    // then
+    assertThat(exceptions).hasSize(0);
+    for (String id : ids) {
+      Job jobResult = managementService.createJobQuery().jobId(id).singleResult();
+      assertThat(jobResult.getRetries()).isEqualTo(RETRIES);
+      assertThat(jobResult.getDuedate()).isEqualTo(newDueDate);
+    }
+  }
+
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_ACTIVITY)
+  @Test
+  public void shouldSetDueDateNull() {
+    // given
+    engineConfiguration.setEnsureJobDueDateNotNull(false);
+    HistoricProcessInstanceQuery historicProcessInstanceQuery = historyService.createHistoricProcessInstanceQuery();
+
+    // assume
+    List<Job> job = managementService.createJobQuery().list();
+    assertThat(job.get(0).getDuedate()).isNull();
+    assertThat(job.get(1).getDuedate()).isNull();
```

Comment:
```suggestion
    assertThat(job.get(0).getDuedate()).isNotNull();
    assertThat(job.get(1).getDuedate()).isNotNull();
```
Whoops. That was a typo. Sorry. Should be `#isNotNull` instead.